android-components/components/browser/state/src/main/java/mozilla/components/browser/state/reducer/TranslationsStateReducer.kt

```diff
@@ -54,40 +55,79 @@ internal object TranslationsStateReducer {
             }
 
         is TranslationsAction.TranslateSuccessAction -> {
-            if (TranslationOperation.TRANSLATE == action.operation) {
-                state.copyWithTranslationsState(action.tabId) {
-                    it.copy(
-                        isTranslated = true,
-                        isTranslateProcessing = false,
-                    )
+            when (action.operation) {
+                TranslationOperation.TRANSLATE -> {
+                    state.copyWithTranslationsState(action.tabId) {
+                        it.copy(
+                            isTranslated = true,
+                            isTranslateProcessing = false,
+                            translationError = null,
+                        )
+                    }
                 }
-            } else {
-                // Restore
-                state.copyWithTranslationsState(action.tabId) {
-                    it.copy(
-                        isTranslated = false,
-                        isRestoreProcessing = false,
-                    )
+
+                TranslationOperation.RESTORE -> {
+                    state.copyWithTranslationsState(action.tabId) {
+                        it.copy(
+                            isTranslated = false,
+                            isRestoreProcessing = false,
+                            translationError = null,
+                        )
+                    }
+                }
+
+                TranslationOperation.FETCH_LANGUAGES -> {
+                    /**
+                     * Generally expect [TranslationsAction.TranslateSetLanguagesAction] to be used
+                     * as the success condition since the updated value is needed.
+                     */
```

Comment:
This is a cranky-old-man comment that should not be actionable unless you're updating the patch for other reasons 😅 :

nit: Javadoc/kdocs are good for API methods, params, constructors, etc. You can use escape comments for implementation docs.

```suggestion
                     // Generally expect [TranslationsAction.TranslateSetLanguagesAction] to be used
                     // as the success condition since the updated value is needed.
```

Reply:
Thank for mentioning the preferred comment pattern, updated!